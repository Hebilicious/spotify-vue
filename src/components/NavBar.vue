<template>
  <div class="side-nav">
    <div class="logo">
      <img :src="Logo" />
      <span>Spotify-Vue</span>
    </div>

    <ul class="nav-links">
      <router-link to="/">
        <span class="material-icons">home</span>
        <span>Home</span>
      </router-link>
      <router-link to="/">
        <span class="material-icons">library_music</span>
        <span>Library</span>
      </router-link>
      <router-link to="/favourite">
        <span class="material-icons">favorite</span>
        <span>Favourites</span>
      </router-link>
    </ul>
  </div>
</template>

<script lang="ts" setup>
import { RouterLink } from 'vue-router';

import Logo from "@/assets/logo.svg";
</script>

<style lang="scss">
.side-nav {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: flex-start;
  grid-area: nav-bar;
  background-color: var(--vt-c-black-dark);
  padding: 1.5rem;
  height: 100%;

  .nav-links {
    list-style: none;
    padding-top: 2rem;
    padding-left: 0;

    a {
      display: flex;
      flex-direction: row;
      align-items: center;
      padding: 0.5rem;
      text-decoration: none;
      color: inherit;
      padding-left: 0;

      &.router-link-active {
        color: var(--color-heading);
      }

      span {
        padding: 0.25rem;
      }

      &:hover {
        color: var(--color-heading);
        cursor: pointer;
      }
    }
  }

  .logo {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;

    img {
      height: 40px;
      max-width: 131px;
    }

    span {
      color: var(--color-heading);
      font-size: 2rem;
      padding-left: 0.5rem;
      letter-spacing: -2px;
    }
  }
}
</style>
