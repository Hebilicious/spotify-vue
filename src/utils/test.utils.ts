import { VueWrapper } from '@vue/test-utils';
import { expect, it } from 'vitest';

export const testSnapShot = (wrapperGetter: () => VueWrapper) => {
  it('Matches the snapshot', () => {
    expect(wrapperGetter().html()).toMatchSnapshot();
  });
};
