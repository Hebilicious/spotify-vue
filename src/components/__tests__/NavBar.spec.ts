import { describe, beforeEach } from 'vitest';
import { shallowMount, VueWrapper } from '@vue/test-utils';

import NavBar from '@/components/NavBar.vue';

import { testSnapShot } from '@/utils/test.utils';

describe('NavBar', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = shallowMount(NavBar);
  });

  testSnapShot(() => wrapper);
});
