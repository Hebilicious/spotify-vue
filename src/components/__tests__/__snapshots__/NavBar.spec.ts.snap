// Vitest Snapshot v1

exports[`NavBar > Matches the snapshot 1`] = `
"<div class="side-nav">
  <div class="logo"><img src="/src/assets/logo.svg?import"><span>Spotify-Vue</span></div>
  <ul class="nav-links">
    <router-link-stub to="/" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
    <router-link-stub to="/" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
    <router-link-stub to="/favourite" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
  </ul>
</div>"
`;
